packages/desktop-client/src/components/reports/spreadsheets/cash-flow-spreadsheet.tsx

```diff
@@ -157,7 +158,13 @@ export function cashFlowByDate(
         })
         .filter({
           $and: [
-            { date: { $transform: '$month', $lte: monthUtils.prevMonth(end) } },
+            {
+              date: {
+                $transform: '$month',
+                $lte: monthUtils.prevMonth(end),
+                $gt: monthUtils.subMonths(end, forecastDivideYears * 12),
```

Comment:
I'm not entirely sure how to contribute to your PR (I can't push to your branch in your repo due to permissions) but you need to split these into seperate objects and combine them with an $and operator.
https://github.com/actualbudget/actual/pull/2674#issuecomment-2087988385

Reply:
@nsulzer , I have the above fix and also the budget source merged in and ready to push. I've created a PR to your branch.